webtau-core/src/main/java/com/twosigma/webtau/expectation/equality/NotEqualMatcher.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2018 TWO SIGMA OPEN SOURCE, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.webtau.expectation.equality;
+
+import com.twosigma.webtau.data.render.DataRenderers;
+import com.twosigma.webtau.expectation.ActualPath;
+import com.twosigma.webtau.expectation.ValueMatcher;
+
+public class NotEqualMatcher implements ValueMatcher {
+    private CompareToComparator comparator;
+    private final Object expected;
+
+    public NotEqualMatcher(Object expected) {
+        this.expected = expected;
+    }
+
+    @Override
+    public String matchingMessage() {
+        return "to not equal " + DataRenderers.render(expected);
+    }
+
+    @Override
+    public String matchedMessage(ActualPath actualPath, Object actual) {
+        return "doesn't equal to " + DataRenderers.render(expected) + "\n" +
+                comparator.generateNotEqualMatchReport();
+    }
+
+    @Override
+    public String mismatchedMessage(ActualPath actualPath, Object actual) {
+        return "equals to " + DataRenderers.render(expected) + ", but shouldn't\n" +
+                comparator.generateNotEqualMismatchReport();
+    }
+
+    @Override
+    public boolean matches(ActualPath actualPath, Object actual) {
+        comparator = CompareToComparator.comparator();
+        return comparator.compareIsNotEqual(actualPath, actual, expected);
+    }
+
+    @Override
+    public String negativeMatchingMessage() {
+        return "to equal " + DataRenderers.render(expected);
+    }
+
+    @Override
+    public String negativeMatchedMessage(ActualPath actualPath, Object actual) {
+        return "equals to " + DataRenderers.render(expected) + "\n" +
+                comparator.generateEqualMatchReport();
+    }
+
+    @Override
+    public String negativeMismatchedMessage(ActualPath actualPath, Object actual) {
+        return "doesn't equal to " + DataRenderers.render(expected) + ", but should\n" +
```

Comment:
"doesn't equal to" sounds a bit weird to me, I think should be either "isn't equal to" or "doesn't equal".  Maybe run it by someone who is an actual native English speaker though, I'm just an approximation to that 😄

Reply:
what about `equal to` above? I am all to make it english as long as it is symmetric.

Reply:
I would say either "is equal to" or "equals"

Reply:
But I'm not really a native speaker so maybe get another opinion.